_episodes/12.md

```diff
@@ -0,0 +1,263 @@
+and let us know there. Otherwise, you can message Manfred Moser or Brian Olsen 
+directly. Also, feel free to reach out to us on our Twitter channels Brian 
+[@bitsondatadev](https://twitter.com/bitsondatadev) and Manfred 
+[@simpligility](https://twitter.com/simpligility).
+
+If you want to show us some 💕, please [give us a ⭐ on Github](/star).
+
+## Release 353 -- Almost
+
+353 is right around the corner. Last show we said this would be a small release.
+While there was a correctness issue we resolved, there didn't seem to be much
+demand to get it out quick as we initially thought. So it was decided to
+continue adding more features to 353. It should be coming out shortly!
+
+
+## Concept of the week: Trino client, Python, and Apache Superset 
+
+What is the general data flow from a connected data source?
+
+ * Trino workers request data from the data source with specific connector
+ * Workers process data and send it to the coordinator
+ * Coordinator does final processing
+ * Supplies the data via HTTP / REST stream to requestor
+ * Requestor is a “client” such as JDBC driver, or Trino CLI
+ * Client translates data further and provides to application (Java application
+   using JDBC driver) or user interface/directly to user (output in CLI)
+ * User views part of data and scrolls down
+ * Client requests more data from coordinator via HTTP / REST (and see above)
+
+What clients are provided by Trino project?
+
+ * JDBC driver
+ * Trino CLI
+ * Trino Python client
+ * Trino Go client
+
+What other clients are there?
+
+ * ODBC driver from Starburst
+ * Various other clients from the open source community 
+  * R
+  * NodeJS/Javascript
+ * See https://trino.io/resources.html#libraries
+
+What happens in the Python world?
+
+Disclaimer: I am not a Pythonista or Pythoneer.
+
+ * DB-API 2.0 
+   * PEP 249 https://www.python.org/dev/peps/pep-0249/
+   * Python standard library
+ * trino-python-client 
+   * Wraps complexity of Trino HTTP / REST
+   * Supports authentication and such
+   * Provides DB API endpoints / implementation
+   * Preferred method to query Trino
+ * SQLAlchemy https://www.sqlalchemy.org/
+   * SQL toolkit
+   * ORM mapper
+   * Widely used, eg. in Apache Superset
+   * Supports dialects
+ * PyHive
+   * Not really a SQL wrapper
+   * Aimed at Hive QL
+   * Only kind of useful for Trino, limited compatibility
+ * JDBC driver (Java !) and PySpark
+   * Possible, but a hack really
+ * PyJDBC 
+   * Wraps DB API around any JDBC driver
+   * Kind of a hack since it goes through JDBC to HTTP, when Trino python
+   client does the same more directly
+ * PyODBC 
+   * Similar hack to PyJDBC
+ * Potentially also possible to talk to via HTTP directly
+   * That’s like reimplementing the trino-python-client
+   * Also see question of the week later
+
+Beyond that, it will vary from application to application.
+
+Let’s find out from our guests how this hangs together in Apache Superset, since
```

Comment:
We need an addition section for the TCB that has something like

Live guests

Jane Done from someorg, twitter url, github url

or something like that

We should at least add our guest from this episode on the text

Reply:
Agreed. Done